flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/extraction/TypeInferenceExtractorTest.java

```diff
@@ -2277,7 +2288,10 @@ public void eval(
     }
 
     private static class MultiStateProcessTableFunction extends ProcessTableFunction<Integer> {
-        public void eval(@StateHint MyFirstState s1, @StateHint MySecondState s2, Integer i) {}
+        public void eval(
+                @StateHint(ttl = "2 days") MyFirstState s1,
+                @StateHint(ttl = "0") MySecondState s2,
+                Integer i) {}
```

Comment:
it seems tests are passing with nanoseconds and microseconds.
However will it be able to handle e2e case with for instance 1 nano ttl?

Reply:
No I think the backend supports only milliseconds. Similar to our timer services as well. This rather sounds like an edge case. Also not sure what happens if you set the global retention time `table.exec.state.ttl` this way.